althea_kernel_interface/src/lib.rs

```diff
@@ -175,13 +176,15 @@ impl KernelInterface {
         ])
     }
 
-    fn read_flow_counters_linux(&mut self) -> Result<Vec<(HwAddr, IpAddr, u64)>, Error> {
+    fn read_flow_counters_linux(&mut self) -> Result<Vec<(MacAddress, IpAddr, u64)>, Error> {
         let output = self.run_command("ebtables", &["-L", "INPUT", "--Lc"])?;
         let mut vec = Vec::new();
         let re = Regex::new(r"-s (.*) --ip6-dst (.*)/.* bcnt = (.*)").unwrap();
         for caps in re.captures_iter(&String::from_utf8(output.stdout)?) {
             vec.push((
-                caps[1].parse::<HwAddr>()?,
+                    MacAddress::parse_str(&caps[1]).unwrap_or_else(|e| {
+                        panic!("{:?}", e);
+                    }), // Ugly and inconsiderate, remove ASAP
```

Comment:
Just wondering, what's the reason here?

Reply:
That MAC package doesn't implement a trait for converting its own error type to standard `Err()`, this seemed quicker to do

Reply:
I'd suggest implementing that trait in the future. Libraries are no place for panics, it's the user who should decide how to handle the parsing error.